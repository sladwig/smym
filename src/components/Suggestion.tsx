import React from 'react';
import './Suggestion.css';
import classnames from 'classnames';
import { useSuggestionStore } from '../zustand/SuggestionStore';

interface SuggestionProps {
    text: string;
    backgroundColor: string;
    index: number;
}
export const Suggestion = ({ text, backgroundColor, index }: SuggestionProps) => {
    const activeIndex = useSuggestionStore(state => state.active);
    const active = index === activeIndex;
    return (
        <div className={classnames('suggestion', { active })}>
            <div className="place-suggestion" style={{ backgroundColor }}>
                {text}
            </div>
        </div>
    );
};
