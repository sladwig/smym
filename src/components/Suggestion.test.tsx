import { render } from '@testing-library/react';
import React from 'react';
import { Suggestion } from './Suggestion';

describe('Suggestion', () => {
    test('renders', () => {
        const { container } = render(<Suggestion />);
        expect(container).toMatchSnapshot();
    });
});
