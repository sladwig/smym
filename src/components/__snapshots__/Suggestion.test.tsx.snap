// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestion renders 1`] = `<div />`;
